backend/app/Services/PayPal/PayPalIPN.php

```
@@ -162,14 +163,23 @@ private function creditUserAccount(string $code, string $uuid): void
         $payment = PayPalDB::getByCode($code);
 
-        User::updateInfo(
-            $token,
-            UserColumns::CREDITS,
-            $currentCredits + $payment['coins'],
-            false
-        );
+        // Execute credit addition with user lock protection to prevent race conditions
+        try {
+            UserLock::executeWithLock($uuid, function () use ($token, $payment) {
+                $currentCredits = User::getInfo($token, UserColumns::CREDITS, false);
```

PHPMD (2.15.0)

Avoid using static access to class '\MythicalDash\Chat\User\User' in method 'creditUserAccount'. (Clean Code Rules)

(StaticAccess)